openquake/hazardlib/gsim/bradley_2013b.py

```diff
@@ -476,3 +479,207 @@ def _compute_additional_sigma(self):
+             + C['sig4'] * AS)
+            # second line
+            * np.sqrt((C['sig3'] * Finferred + 0.7 * Fmeasured)
+                      + (1 + NL) ** 2)
+        )
+        # Get sigma reduction factors if site is in CBD polygon.
+        srf_sigma = np.array(np.ones(np.shape(in_cbd)))
+        srf_phi = np.array(np.ones(np.shape(in_cbd)))
+        srf_tau = np.array(np.ones(np.shape(in_cbd)))
+        if in_cshm is True:
+            srf_sigma[in_cbd == True] = self._get_SRF_sigma(imt_per)
+            srf_phi[in_cbd == True] = self._get_SRF_phi(imt_per)
+            # The tau reduction term is not used in this implementation
+            # srf_tau[in_cbd == True] = self._get_SRF_tau(imt_per)
+
+        # Add 'additional sigma' specified in the Canterbury Seismic
+        # Hazard Model to total sigma
+        additional_sigma = self._compute_additional_sigma()
+
+        ret = []
+        for stddev_type in stddev_types:
+            assert stddev_type in self.DEFINED_FOR_STANDARD_DEVIATION_TYPES
+            if stddev_type == const.StdDev.TOTAL:
+                # eq. 21
+                scaled_sigma = np.sqrt(((1 + NL) ** 2) *
+                                       (tau ** 2) + (sigma ** 2)) * srf_sigma
+                ret += [np.sqrt(scaled_sigma ** 2 + additional_sigma ** 2)]
+            elif stddev_type == const.StdDev.INTRA_EVENT:
+                scaled_phi = sigma * srf_phi
+                ret.append(np.sqrt(
+                    scaled_phi ** 2 + additional_sigma ** 2 / 2))
+            elif stddev_type == const.StdDev.INTER_EVENT:
+                # this is implied in eq. 21
+                scaled_tau = (np.abs((1 + NL) * tau)) * srf_tau
+                ret.append(np.sqrt(
+                    scaled_tau ** 2 + additional_sigma ** 2 / 2))
+
+        return ret
+
+    def _adjust_mean_model(self, in_cshm, in_cbd, imt_per, b13_mean):
+
+        dL2L = dS2S = np.array(np.zeros(np.shape(b13_mean)))
+        # If the site is in the CBD polygon, get dL2L and dS2S terms
+        if in_cshm is True:
+            # Only apply the dL2L term only to sites located in the CBD.
+            dL2L[in_cbd == True] = self._get_dL2L(imt_per)
+            dS2S[in_cbd == True] = self._get_dS2S(imt_per)
+
+        mean = b13_mean + dL2L + dS2S
+
+        return mean
+
+    def _check_in_cbd_polygon(self, lons, lats):
+        """
+        Checks if site is located within the CBD polygon. The boundaries of
+        the polygon implemented here are from the 'Central City' Zoning Map
+        in the Christchurch District Plan. See Figure 4.4 of Van Houtte and
+        Abbott (2019).
+        """
+        polygon = shapely.geometry.Polygon(
+            [(172.6259, -43.5209), (172.6505, -43.5209),
+             (172.6505, -43.5399), (172.6124, -43.5400),
+             (172.6123, -43.5289), (172.6124, -43.5245),
+             (172.6220, -43.5233)]
+        )
+        points = [shapely.geometry.Point(lons[ind], lats[ind])
```

Comment:
For most applications it is not a good idea to be doing point-in-polygon checks for Earth geometry using geodetic coordinates, as the edge of a polygon should be a great arc and not a line in long/lat. In this case though, the area is very small so maybe the error is tolerable given the overhead in re-projection.

Reply:
Agreed. The other point-in-polygon function (_check_in_cshm_polygon) was how the original Canterbury model was defined, but I don't see why that can't be changed. More than happy to amend. 

Is there anything else that jumps out?

Reply:
Sorry, it's a different reviewer this time around! Everything else looks good to me.